src/System.Private.Reflection.Execution/src/Internal/Reflection/Execution/ExecutionEnvironmentImplementation.MappingTables.cs

```diff
@@ -1089,17 +1089,45 @@ private unsafe bool TryGetMethodForOriginalLdFtnResult_Inner(NativeFormatModuleI
                         {
                             unsafe
                             {
-                                if (fieldAccessMetadata.Flags.HasFlag(FieldTableFlags.IsGcSection))
-                                    fieldAddress = *(IntPtr*)TypeLoaderEnvironment.Instance.TryGetGcStaticFieldData(declaringTypeHandle) + fieldAccessMetadata.Offset;
+                                if ((fieldAccessMetadata.Flags & FieldTableFlags.IsGcSection) != 0)
+                                {
+                                    IntPtr gcStaticsBase = *(IntPtr*)TypeLoaderEnvironment.Instance.TryGetGcStaticFieldData(declaringTypeHandle);
+#if CORERT
+                                    // The GC statics are allocated at runtime instead of being stored in the image.
+                                    // A double indirection is needed to get the the GC statics base allocated by the runtime.
+                                    gcStaticsBase = **(IntPtr**)gcStaticsBase;
```

Comment:
That is a good point! Yes I think we'll need something to account for possible garbage collections